src/query/storages/fuse/src/operations/common/snapshot_generator.rs

```diff
@@ -59,11 +61,32 @@ pub trait SnapshotGenerator {
     ) -> Result<TableSnapshot>;
 }
 
+#[derive(Clone, serde::Serialize, serde::Deserialize, Debug, PartialEq)]
+pub struct DeleteConflictResolveContext {
+    pub removed_segments: Vec<Location>,
+    pub added_segments: Vec<Location>,
+    pub removed_statistics: Statistics,
+    pub added_statistics: Statistics,
+}
+
+#[derive(Clone, serde::Serialize, serde::Deserialize, Debug, PartialEq, Default)]
+pub enum ConflictResolveContext {
+    Delete(Box<DeleteConflictResolveContext>),
+    Compact,
+    Update,
+    Replace,
+    Recluster,
+    Insert,
+    #[default]
+    Uninitialized,
```

Comment:
is it possible to assign a non-`Uninitialized` resolve context to `MutationGenerator`? so that some relevant `unreachable` match branches seem to be avoidable?  

or postponing the refactoring to another dedicated PR is also fine.

Reply:
Good idea, I will refactor in another PR